newm-server/src/main/kotlin/io/newm/server/features/song/AgreementMessageReceiver.kt

```diff
@@ -0,0 +1,45 @@
+package io.newm.server.features.song
+
+import com.amazonaws.services.sqs.model.Message
+import io.ktor.server.application.*
+import io.ktor.util.logging.*
+import io.newm.server.aws.SqsMessageReceiver
+import io.newm.server.di.inject
+import io.newm.server.ext.toUUID
+import io.newm.server.features.song.model.AgreementMessage
+import io.newm.server.features.song.model.MintingStatus
+import io.newm.server.features.song.model.Song
+import io.newm.server.features.song.repo.SongRepository
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.slf4j.MarkerFactory
+
+class AgreementMessageReceiver : SqsMessageReceiver {
+    private val repository: SongRepository by inject()
+    private val logger: Logger by inject()
+    private val marker = MarkerFactory.getMarker(javaClass.simpleName)
+
+    @OptIn(ExperimentalSerializationApi::class)
+    private val json = Json {
```

Comment:
Can this be a Koin object that gets injected. I don't think there's a need to re-create Json objects everywhere unless they have some custom configuration needed.

Reply:
Good call! I'll do that.

Reply:
Great idea. Done!